site/documentation/content/api/management/device-registry-v1.yaml

```diff
@@ -461,7 +487,72 @@ components:
          description: Allows arbitrary properties as extension to the ones
                       specified by the Hono API.
 
-# Tenant schema
+
+      PagingSortingFilteringOptions:
+         type: object
+         additionalProperties: false
+         properties:
+            "page":
+               $ref: '#/components/schemas/PagingOptions'
+            "filters":
+               $ref: '#/components/schemas/FilterOptionsSet'
+            "sort":
+               $ref: '#/components/schemas/SortingOptions'
+
+      PagingOptions:
+         type: object
+         additionalProperties: false
+         description: options to page the results of bulk queries.
+         properties:
+            "limit":
+               type: integer
+               default: 30
+               minimum: 0
+               maximum: 200
+               description: |
+                  The threshold on the number of results returned in a single API call. If not set, the registry will return up to 30 results.
+            "offset":
+               type: integer
+               default: 0
+               minimum: 0
+               description: |
+                  The offset to use as a starting point to return results. This allows to go through the query results in
+                  a paged manner when the query returns more results than the limit.
+
+      FilterOptionsSet:
+         type: array
+         description: A set of filter options. If multiples filters are provided, they will be applied together
+         in a cumulative way.
+         items:
+            $ref: '#/components/schemas/FilterOption'
+
+      FilterOption:
+            type: object
+            additionalProperties: true
+            description: Allows arbitrary filters to be applied to bulk queries.
+                         Fields to filter MUST be designated with their JSON Path representation.
```

Comment:
I think it would be better to use the same approach as we do for the sorting options, i.e. explicitly define two properties one containing the *path* expression and the other containing the value to match. That way we would be able to define constraints on the properties (if we so choose) and the list of filters actually makes some sense because otherwise there is no need for an array of FilterOption objects because a single FilterObject could already contain arbitrary filter expressions ...

Reply:
Yes it's a good idea to have sorting options and filters to be designed the same way :)